flask_resources/views/singleton.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Flask-Resources is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Flask Resources module to create REST APIs."""
+
+from ..args.parsers import item_request_parser
+from .base import BaseView, with_resource_requestctx
+
+
+class SingletonView(BaseView):
+    """Parse request args and create a request context.
+
+    Note that the resource route should contain the `<id>` param.
+    """
+
+    def __init__(
+        self, resource=None, item_parser=item_request_parser, *args, **kwargs
+    ):
+        super(SingletonView, self).__init__(resource=resource, *args, **kwargs)
+        self.item_parser = item_parser
```

Comment:
Ideally the parsing should be moved to `flask-resources`. However, it is not the easiest thing to do now. I can make it a `TODO`. Due to time constraints I do not think this is a huge inconvenient since it has a default value.

Worst case it is not used...